lib/grn_ecmascript.lemon

```diff
@@ -395,21 +399,27 @@ relational_expression ::= relational_expression NEAR shift_expression. {
     grn_obj *max_element_intervals;
     GRN_PTR_POP(&efsi->max_element_intervals_stack,
                 max_element_intervals);
-    if (max_element_intervals) {
-      grn_expr_append_obj(efsi->ctx, efsi->e, max_element_intervals,
-                          GRN_OP_PUSH, 1);
-      grn_expr_take_obj(efsi->ctx, efsi->e, max_element_intervals);
-      n_args++;
+    if (!max_element_intervals) {
+      break;
     }
-  }
+    grn_expr_append_obj(efsi->ctx, efsi->e, max_element_intervals,
+                        GRN_OP_PUSH, 1);
+    grn_expr_take_obj(efsi->ctx, efsi->e, max_element_intervals);
+    n_args++;
+    int min_interval;
+    GRN_INT32_POP(&efsi->min_interval_stack, min_interval);
+    grn_expr_append_const_int(efsi->ctx, efsi->e, min_interval,
+                              GRN_OP_PUSH, 1);
+    n_args++;
+  } while(0);
```

Comment:
We can't use `exit:` label and `goto` in each `relational_expression ::= relational_expression XXXX shift_expression.`s because they are extended to the same function, and that causes multiple label definition as a result.
So I adopted the `do { ... } while(0)` approach.